src/Features/Core/Portable/PullMemberUp/QuickAction/AbstractMemberPullerWithQuickAction.cs

```diff
@@ -29,8 +34,165 @@ internal abstract class AbstractMemberPullerWithQuickAction
+            {
+                return new SolutionChangeAction(
+                    string.Format(FeaturesResources.Add_to_0, result.Destination),
+                    token => PullMembersUpAsync(result, contextDocument, destinationNodeSyntax, token));
+            }
+            else
+            {
+                throw ExceptionUtilities.UnexpectedValue(result.Destination);
+            }
+        }
+
+        private async Task<Solution> PullMembersUpAsync(
+            PullMembersUpAnalysisResult result,
+            Document contextDocument,
+            SyntaxNode destinationNodeSyntax,
+            CancellationToken cancellationToken)
+        {
+            var solution = contextDocument.Project.Solution;
+            var solutionEditor = new SolutionEditor(solution);
+            var codeGenerationService = contextDocument.Project.LanguageServices.GetRequiredService<ICodeGenerationService>();
+            var (editorMap, syntaxMap) = await InitializeEditorMapsAndSyntaxMapAsync(result, destinationNodeSyntax, solutionEditor, solution, cancellationToken).ConfigureAwait(false);
+            return PullMembersIntoClass(
+                result, codeGenerationService, destinationNodeSyntax,
+                solutionEditor, solution, editorMap, syntaxMap);
+        }
+
+        private CodeAction PullMembersIntoInterface(
+            PullMembersUpAnalysisResult result,
+            Document contextDocument,
+            ICodeGenerationService codeGenerationService)
+        {
+            var symbolsToPullUp = result.MembersAnalysisResults.
+                SelectAsArray(analysisResult =>
+                {
+                    if (analysisResult.Member is IPropertySymbol propertySymbol)
+                    {
+                        // When it is a property, it could have a public getter/setter
+                        // but other one is not public. In this scenario, only the public getter/setter
+                        // will be add to the destination interface
+                        return CodeGenerationSymbolFactory.CreatePropertySymbol(
+                            propertySymbol,
+                            getMethod: FilterGetterOrSetter(propertySymbol.GetMethod),
+                            setMethod: FilterGetterOrSetter(propertySymbol.SetMethod));
+                    }
+                    else
+                    {
+                        return analysisResult.Member;
+                    }
+                });
+            return new DocumentChangeAction(
+                string.Format(FeaturesResources.Add_to_0, result.Destination),
+                cancellationToken =>
+                {
+                    var options = new CodeGenerationOptions(generateMethodBodies: false, generateMembers: false);
+                    return codeGenerationService.AddMembersAsync(
+                        contextDocument.Project.Solution, result.Destination, symbolsToPullUp, options: options, cancellationToken: cancellationToken);
```

Comment:
it's a good idea to extract this code to a method so it doesn't accidentally capture anything it shouldn't.